projects/RabbitMQ.Client/client/impl/ContentHeaderPropertyWriter.cs

```diff
@@ -45,92 +45,101 @@
 
 namespace RabbitMQ.Client.Impl
 {
-    struct ContentHeaderPropertyWriter
+    internal ref struct ContentHeaderPropertyWriter
     {
-        private int _bitCount;
-        private ushort _flagWord;
-        public int Offset { get; private set; }
-        public Memory<byte> Memory { get; private set; }
+        private const ushort StartBitMask = 0b1000_0000_0000_0000;
+        private const ushort EndBitMask = 0b0000_0000_0000_0001;
 
-        public ContentHeaderPropertyWriter(Memory<byte> memory)
+        private readonly Span<byte> _span;
+        private int _offset;
+        private ushort _bitAccumulator;
+        private ushort _bitMask;
+
+        public int Offset => _offset;
+
+        private Span<byte> Span => _span.Slice(_offset);
+
+        public ContentHeaderPropertyWriter(Span<byte> span)
         {
-            Memory = memory;
-            _flagWord = 0;
-            _bitCount = 0;
-            Offset = 0;
+            _span = span;
+            _offset = 0;
+            _bitAccumulator = 0;
+            _bitMask = StartBitMask;
         }
 
         public void FinishPresence()
         {
-            EmitFlagWord(false);
+            WriteBits();
         }
 
         public void WriteBit(bool bit)
         {
             WritePresence(bit);
         }
-
+        
```

Comment:
There are several spots with extra whitespace like this.

Reply:
wrong vs setting, occurrences should be fixed by 063e604